contracts/Core/RewardManager.sol

```diff
@@ -98,16 +90,25 @@ contract RewardManager is Initializable, ACL, RewardStorage {
         uint256 epochLastActive = thisStaker.epochStaked < thisStaker.epochLastRevealed
             ? thisStaker.epochLastRevealed
             : thisStaker.epochStaked;
+
         // penalize or reward if last active more than epoch - 1
         uint256 inactiveEpochs = (epoch - epochLastActive == 0) ? 0 : epoch - epochLastActive - 1;
 
         if (inactiveEpochs <= parameters.gracePeriod()) {
             return;
         }
         uint256 previousStake = thisStaker.stake;
+        uint256 newStake = thisStaker.stake;
         uint256 previousAge = thisStaker.age;
+        uint256 newAge = thisStaker.age;
+
+        // Not reveal penalty due to Randao
+        if (thisStaker.epochLastRevealed < thisStaker.epochLastCommitted) {
```

Comment:
Hey 
Is there a scenario 
Where it would be best for staker to commit but not reveal?

Reply:
Yes.
Attacker waits till everyone has revealed.
then only reveals if his reveal will make him the block proposer.

Reply:
Oh yes